packages/material-ui/src/test-utils/describeConformance.js

```diff
@@ -0,0 +1,140 @@
+ *                                                 be rendered instead.
+ */
+function testComponentProp(element, options) {
+  const { testComponentPropWith: component = 'em', mount } = options;
+
+  it('can render another root component with the `component` prop', () => {
+    const wrapper = mount(React.cloneElement(element, { component }));
+
+    assert.strictEqual(findOutermostIntrinsic(wrapper).type(), component);
+  });
+}
+
+/**
+ * Material-UI components can spread additional props to a documented component.
+ * It's set via @inheritComponent in the source.
+ *
+ * @param {React.ReactElement} element
+ * @param {Object} options
+ * @param {string} options.inheritComponentName - The element type display name
+ *                                                that receives spread props.
+ */
+function testPropsSpread(element, options) {
+  const { inheritComponentName, mount } = options;
+
+  it(`does spread props to the first ${inheritComponentName}`, () => {
+    const testProp = 'data-test-props-spread';
+    const value = randomStringValue();
+    const wrapper = mount(React.cloneElement(element, { [testProp]: value }));
+    assert.strictEqual(
+      wrapper
+        .find(inheritComponentName)
+        .first()
+        .props()[testProp],
+      value,
+    );
+  });
+}
+
+/**
+ * Some Material-UI components can forward their refs via `innerRef`
+ *
+ * TODO: Use `ref` once WithStylesTest is no longer required
+ *
+ * @param {React.ReactElement} element
+ * @param {Object} options
+ * @param {boolean} options.forwardRef - If `true` then this component forwards `ref`
+ */
+function testRefForwarding(element, options) {
+  const { forwardRef, mount } = options;
+
+  describe('prop: innerRef', () => {
+    before(() => {
+      // just to swallow possible error messages from attaching refs to function components
+      consoleErrorMock.spy();
+    });
+
+    after(() => {
+      consoleErrorMock.reset();
+    });
+
+    it(`does ${!forwardRef ? 'not' : ''} forward ref`, () => {
+      const ref = React.createRef();
+      mount(React.cloneElement(element, { innerRef: ref }));
+
+      if (!forwardRef) {
+        assert.strictEqual(ref.current, null);
+      } else {
+        assert.ok(ref.current);
+      }
+    });
+  });
+}
+
+const fullSuite = {
+  class: testClassName,
+  componentProp: testComponentProp,
+  propsSpread: testPropsSpread,
+  refForwarding: testRefForwarding,
+};
+
+/**
+ * Tests various aspects of a component that should be equal across Material-UI
+ * components.
+ *
+ * @param {React.ReactElement} minimalElement - the component with it's minimal required props
+ * @param {Object} options
+ * @param {string} options.classes - see testClassName
+ * @param {string} options.inheritComponentName - see testPropsSpread
+ * @param {function} options.mount - Should be a return value from createMount
+ * @param {boolean} options.noForwardRef - see test testRefForwarding
```

Comment:
forwardRef

Reply:
Please use GitHub Suggestions :pray:

Reply:
I was on my phone 👼

Reply:
```suggestion
 * @param {boolean} options.forwardRef - see test testRefForwarding
```

Reply:
> Please use GitHub Suggestions 🙏

Please accept suggestions using tool provided by GitHub, rather than pushing a new commit. 😈

Reply:
Why? I'd rather save my time. One click is easier than checking out, searching the file, making change, rebasing (like that would matter if we switch anywy), pushing and not having the original attribution present.

Reply:
Seems you are arguing against yourself in that case...

Reply:
Guys, it's a simple typo, let's move forward.